src/settings/mod.rs

```diff
@@ -355,11 +356,18 @@ impl From<&Configuration> for RuleTable {
             }
         }
 
-        // Validate that we didn't enable any incompatible rules.
-        for (a, b, message) in INCOMPATIBLE_CODES {
-            if rules.enabled(a) && rules.enabled(b) {
-                warn_user_once!("{}", message);
-            }
+        // Validate that we didn't enable any incompatible rules. Use this awkward approach to
+        // give each pair it's own `warn_user_once`.
+        let [first, second] = INCOMPATIBLE_CODES;
+        let (preferred, expendable, message) = first;
+        if rules.enabled(preferred) && rules.enabled(expendable) {
+            warn_user_once!("{}", message);
+            rules.disable(expendable);
+        }
+        let (preferred, expendable, message) = second;
+        if rules.enabled(preferred) && rules.enabled(expendable) {
+            warn_user_once!("{}", message);
+            rules.disable(expendable);
```

Comment:
Previously, we were only warning for the _first_ incompatible rule pair, so users never saw the `D212` / `D213` warning!